src/ui/generic-components/TrafficLightButton.swift

```diff
@@ -1,219 +1,379 @@
         super.init(frame: .init(origin: .zero, size: .init(width: size, height: size)))
         self.type = type
-        target = self
-        action = #selector(onClick)
+
+        // Adjust the button size to fit the specified dimensions
         fit(size, size)
+        // Add a tracking area to detect mouse enter and exit events
         addTrackingArea(NSTrackingArea(rect: bounds, options: [.mouseEnteredAndExited, .activeInKeyWindow], owner: self, userInfo: nil))
         toolTip = tooltip
+        wantsLayer = true
     }
 
+    /// Required initializer for decoding the button (not used here).
     required init?(coder: NSCoder) {
         super.init(coder: coder)
     }
 
-    @objc func onClick() {
-        if (type == .fullscreen) {
-            window_?.toggleFullscreen()
-        } else if (type == .miniaturize) {
-            window_?.minDemin()
-        } else if (type == .close) {
-            window_?.close()
-        } else if (type == .quit) {
-            window_?.application.quit()
-        }
-    }
-
+    /// Mouse entered event handler.
     override func mouseEntered(with event: NSEvent) {
         isMouseOver = true
-        setNeedsDisplay()
+        setNeedsDisplay(bounds)
     }
 
+    /// Mouse exited event handler.
     override func mouseExited(with event: NSEvent) {
         isMouseOver = false
-        setNeedsDisplay()
+        setNeedsDisplay(bounds)
+    }
+
+    /// Mouse up event handler.
+    override func mouseUp(with event: NSEvent) {
+        if event.clickCount >= 1 {
+            performClick()
+        }
+        setNeedsDisplay(bounds)
+    }
+
+    /// Perform the button click action.
+    private func performClick() {
+        if let window = targetWindow, let type = type {
+            switch type {
+            case .fullscreen:
+                window.toggleFullscreen()
+            case .miniaturize:
+                window.minDemin()
+            case .close:
+                window.close()
+            case .quit:
+                window.application.quit()
+            }
+        }
     }
 
+    /// Custom drawing of the button.
+    ///
+    /// - Parameters:
+    ///   - dirtyRect: The portion of the view’s bounds that needs to be redrawn.
     override func draw(_ dirtyRect: NSRect) {
+        super.draw(dirtyRect)
+        // Retrieve the colors for the button based on its type and system control tint
         let (diskBackgroundColor, diskStrokeColor, symbolColor) = colors()
-        let disk = drawDisk(diskBackgroundColor, diskStrokeColor)
+
+        let disk = NSBezierPath()
+        disk.appendOval(in: NSMakeRect(bounds.origin.x + 0.5, bounds.origin.y + 0.5, bounds.width - 1, bounds.height - 1))
+
+        // Fill the disk with the black background color
+        NSColor.black.setFill()
+        disk.fill()
+
+        // Fill the disk with the background color
+        diskBackgroundColor.draw(in: disk, relativeCenterPosition: .zero)
```

Comment:
It seems you added drawing a black disk here. What issue is it fixing?

Reply:
This is the key to fixing the issue of transparent button in the dark theme. The black disk serves as a background to prevent the buttons from being transparent.